build.cake

```diff
@@ -116,11 +122,24 @@ Task("Build")
     .IsDependentOn("NuGetRestore")
     .Does(() =>
     {
-        DotNetBuild(SOLUTION_PATH, settings => settings
-            .WithTarget("Build")
-            .SetConfiguration(configuration)
-            .SetVerbosity(Verbosity.Minimal)
-        );
+		if(IsRunningOnWindows())
+		{
+			MSBuild(SOLUTION_FILE, new MSBuildSettings()
+				.SetConfiguration(configuration)
+				.SetMSBuildPlatform(MSBuildPlatform.Automatic)
+				.SetVerbosity(Verbosity.Minimal)
+				.SetNodeReuse(false)
+				.SetPlatformTarget(PlatformTarget.MSIL)
+			);
+		}
+		else
+		{
+			XBuild(SOLUTION_FILE, new XBuildSettings()
+				.WithTarget("Build")
+				.WithProperty("Configuration", configuration)
+				.SetVerbosity(Verbosity.Minimal)
+			);
+		}
```

Comment:
I get constant errors with DotNetBuild, so I went to the approach we use for the framework.